src/main/java/com/hazelcast/aws/EcsClientStrategy.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.aws;
+
+import com.hazelcast.aws.impl.Constants;
+import com.hazelcast.aws.impl.DescribeNetworkInterfacesOperation;
+import com.hazelcast.aws.impl.DescribeTasksOperation;
+import com.hazelcast.aws.impl.Ec2OperationClient;
+import com.hazelcast.aws.impl.EcsOperationClient;
+import com.hazelcast.aws.impl.ListTasksOperation;
+import com.hazelcast.aws.utility.Environment;
+import com.hazelcast.aws.utility.MetadataUtils;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.hazelcast.aws.impl.Constants.AWS_EXECUTION_ENV_VAR_NAME;
+import static com.hazelcast.aws.impl.Constants.EC2_PREFIX;
+import static com.hazelcast.aws.utility.StringUtils.isNotEmpty;
+import static java.lang.String.format;
+
+/**
+ * Strategy for discovery of Hazelcast instances running under ECS / Fargate
+ */
+class EcsClientStrategy extends AwsClientStrategy {
+
+    private static final String UPPER_ECS = "ECS";
+
+    private static final ILogger LOGGER = Logger.getLogger(EcsClientStrategy.class);
+
+    private String endpointDomain;
+
+    EcsClientStrategy(AwsConfig awsConfig, String endpoint) {
+        super(awsConfig, endpoint);
+        this.endpointDomain = endpoint.substring(Constants.HOSTNAME_PREFIX_LENGTH);
+    }
+
+    @Override
+    @SuppressWarnings(value = "unchecked")
+    public Map<String, String> getAddresses() throws Exception {
+        Map<String, String> metadata = retrieveAndParseMetadata();
+        String metadataClusterName = metadata.get("clusterName");
+        String metadataFamilyName = metadata.get("familyName");
+        EcsOperationClient ecsOperationClient = new EcsOperationClient(awsConfig, endpoint);
+        Ec2OperationClient ec2OperationClient = new Ec2OperationClient(awsConfig, EC2_PREFIX + endpointDomain);
+        Collection<String> taskArns = ecsOperationClient.execute(new ListTasksOperation(metadataClusterName, metadataFamilyName));
```

Comment:
Correct me if I'm wrong, but I believe this behavior will limit hazelcast members to only those within the same ECS cluster.  I specifically create separate ECS clusters for different use cases and communicate just fine today across ECS cluster boundaries on the 2.2 version of the plugin (plus some network config on ECS and in Java) so to do this would be a step backwards in my opinion.  Why would you not also search the tag-type and tag-value attributes in EC2 for other Hazelcast cluster members instead of introducing such a limitation?

Reply:
Alternatively to searching all EC2 instances you could add a config param to allow the user to specify the cluster names he/she wishes to search for tasks.

Reply:
Thanks for the comments @markathomas, interesting use case.
I believe searching EC2 tags is something that wouldn't fit the Fargate model (though it'd be appropriate for ECS on EC2). 
I really like your suggestion to specify multiple cluster names actually! I'll put that on my backlog for this PR :-)

Cheers,
Guglielmo

Reply:
I guess the approach mentioned lines up with this doc : https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cmd-ecs-cli-ps.html you can find all ips of running task in you cluster (whatever is Fargate or not) so it should be possible use the same api call in hazelcast to find the ips

Reply:
With new instance ids etc., you can propagate tags to alle fargate tasks through a service. So looking up tasks by tag would be a convinient clustering mechanism i think.

Ref. your comment @googlielmo